// Copyright (c) farm-ng, inc.
//
// Licensed under the Amiga Development Kit License (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://github.com/farm-ng/amiga-dev-kit/blob/main/LICENSE
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

import "farm_ng/core/timestamp.proto";

package farm_ng.service.proto;

service Service {
  rpc getServiceState(GetServiceStateRequest) returns (GetServiceStateReply) {}
}

// TODO: Just use a bool where this is used?
enum ReplyStatus {
  OK = 0;
  FAILED = 1;
}

enum ServiceState {
  // Service state is not known
  UNKNOWN = 0;
  // Running, and actively serving data
  RUNNING = 1;
  // Idle, waiting for clients
  IDLE = 2;
  // Service isn't running at all
  UNAVAILABLE = 3;
  // Service is in some error state
  ERROR = 4;
}

message GetServiceStateRequest {
}

message GetServiceStateReply {
  ServiceState state = 1;
  uint32 pid = 2; // process id
  double uptime = 3;
  repeated farm_ng.core.proto.Timestamp timestamps = 4;
}
